### PR TITLE
[Resources] Use JsonExtensions to serialize and deserialize template and parameter objects

### PR DESCRIPTION
**Pull request by jorgecotillo**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Resources/Resources.Test/Resources.Test.csproj [hunk 1]
```diff
@@ -21,6 +21,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\ResourceManager\ResourceManager.csproj" />
```
Comment (jorgecotillo) [inline src/Resources/Resources.Test/Resources.Test.csproj:24]: Added this reference to be able to access JsonExtensions, probably, as a future PR, consider moving JsonExtensions to a common project.

**Review by Xynoclafe — DISMISSED** [review]

**Review by isra-fel — APPROVED** [review]
LGTM

**Comment by jorgecotillo** [comment]
Just fixed a merge conflict in ChangeLog.md, do we have an ETA when this fix will be merged and after is merged how long it takes for it to be published? Thanks!

**Review by isra-fel — APPROVED** [review]

**Comment by isra-fel** [comment]
Next official release will be September 7

**Comment by ChristopherGLewis** [comment]
Did this break Tag case sensitivity?

https://github.com/Azure/azure-powershell/issues/15876

**Pull request merged**